File: client/src/types/expense.ts
export interface Expense {
  id: number;
  description: string;
  amount: string | number;
  date: string;
  category: string;
}

export interface CreateExpenseRequest {
  description: string;
  amount: string | number;
  date: string;
  category: string;
}

export interface CategorySummary {
  category: string;
  total: number | string;
}

File: server/src/main/java/com/example/server/entity/Expense.java
package com.example.server.entity;

import jakarta.persistence.*;
import lombok.*;
import com.fasterxml.jackson.annotation.JsonIgnore;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "expenses")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Expense {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String description;

    @Column(nullable = false, precision = 19, scale = 2)
    private BigDecimal amount;

    @Column(nullable = false)
    private LocalDate date;

    @Column(nullable = false)
    private String category;

    // NEW: Link to User
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnore // Prevent circular reference in JSON
    private User user;
}
